hazelcast/src/test/java/com/hazelcast/internal/partition/MigrationInvocationsSafetyTest.java

```diff
@@ -149,4 +150,47 @@ public void run() {
         });
 
     }
+
+    @Test
+    public void partitionState_shouldNotBeSafe_duringPartitionTableFetch_whenMasterLeaves() {
+        partitionState_shouldNotBeSafe_duringPartitionTableFetch_whenMasterChanges(false);
+    }
+
+    @Test
+    public void partitionState_shouldNotBeSafe_duringPartitionTableFetch_whenLiteMasterLeaves() {
+        partitionState_shouldNotBeSafe_duringPartitionTableFetch_whenMasterChanges(true);
+    }
+
+    private void partitionState_shouldNotBeSafe_duringPartitionTableFetch_whenMasterChanges(boolean liteMaster) {
+        final HazelcastInstance initialMaster = factory.newHazelcastInstance(new Config().setLiteMember(liteMaster));
+        final HazelcastInstance nextMaster = factory.newHazelcastInstance();
+        final HazelcastInstance slave = factory.newHazelcastInstance();
+
+        assertClusterSizeEventually(3, nextMaster, slave);
```

Comment:
Any reason that `initialMaster` is not part of the list? Could be replaced with
```java
        assertClusterSizeEventually(3, factory.getAllHazelcastInstances());
        warmUpPartitions(factory.getAllHazelcastInstances());
```

Reply:
Same here. Master will know the latest cluster formation before rest of the cluster.